pkg/config/exec_config.go

```diff
@@ -0,0 +1,67 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/warptools/warpforge/pkg/formulaexec"
+	"github.com/warptools/warpforge/pkg/plotexec"
+	"github.com/warptools/warpforge/pkg/workspace"
+)
+
+func BinPath(state State) string {
+	// determine the path of the running executable
+	// other binaries (runc, rio) will be located here as well
+	path, ok := state.Env[EnvWarpforgePath]
+	if ok {
+		return path
+	}
+	return filepath.Dir(state.ExecutablePath)
+}
+
+func KeepRunDir(state State) bool {
+	_, ok := state.Env[EnvWarpforgeKeepRundir]
+	return ok
+}
+
+func RunPathBase(state State) string {
+	if value, ok := state.Env[EnvWarpforgeRunPath]; ok {
+		return value
+	}
+	return state.TempDir
+}
+
+func WarehousePathOverride(state State) *string {
+	value, ok := state.Env[EnvWarpforgeWarehouse]
+	if !ok {
+		return nil
+	}
+	return &value
+}
+
+// Retrieves workspace stack at state.WorkingDirectory
+// Errors:
+//
+//  - warpforge-error-searching-filesystem -- unexpected error traversing filesystem
+func DefaultWorkspaceStack(state State) (workspace.WorkspaceSet, error) {
```

Comment:
Think I'd probably make this a sibling of the `workspace.FindWorkspaceStack` in that package, while still accepting a config object as a parameter.

Reply:
I'm going to remove it and leave it up to packages to create a "pwd+wss" utility function where needed.